### DISCUSSION TITLE
Flutter - Using webview widget to connect with AptosConnect wallet

### DISCUSSION THREAD

**Discussion by dbrdz** [aptos-labs/aptos-developer-discussions#536] [Questions] [answered]
### Describe your question in detail.

I'm trying to connect to the aptosconnect wallet in my flutter web3 app. I'm using a Webview widget to open up the aptosconnect page and start the connection but I can't figure out how to build the request that goes in the url as a query parameter. 

Is there some sort of documentation as to how to communicate with the aptosconnect wallet?

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

### Describe your environment or tooling in detail

**Operating system:** MacOS
**Front-end framework:** Flutter

**Comment by 0xbrw** [comment]
cc @darren-wangg who has more context on aptos connect

**Comment by darren-wangg** [comment]
We don't offer native support for Dart so you would have to communicate with our JS package. The easiest way to do that is using the [Aptos Wallet Adapter](https://github.com/aptos-labs/aptos-wallet-adapter/tree/main/packages/wallet-adapter-react) (docs [here](https://aptosconnect.app/docs/quick-start/)).

I have limited experience with Flutter, so don't have a good answer on how to use a Webview to execute Javascript code. But if you find a way to run JS within your app, using that package should be pretty seamless.

You can also try to ask in our Telegram group if anyone else has used Flutter before: https://t.me/+PTnfbYjGpqMxMDlh

**Comment by dbrdz** [comment, marked as answer]
Thank you for the quick response! I've been playing around with the web view widget in flutter and I was able to go through the connect request.

I'll leave the code I used in case someone wants to experiment with this too:

https://github.com/BananaQuery/aptosconnect_flutter
